### PR TITLE
Nsbf API

### PR DESCRIPTION
**Pull request by RoadRunnr**
This contains fragments from the coveralls branch, Will be rebased once that branch is merged.

### CODE REVIEW

**Comment by RoadRunnr** [comment]
> This contains fragments from the coveralls branch, Will be rebased once that branch is merged.

scratch that. rebased to master without coveralls changes

File: include/ergw.hrl [hunk 1]
```diff
@@ -149,3 +149,12 @@
 	  rx :: {Bytes :: integer(), Packets :: integer()},
 	  tx :: {Bytes :: integer(), Packets :: integer()}
 	 }).
+
+%% nBsf registration record
+-record(bsf, {
+	  dnn                       :: [binary()],
+	  snssai = {1, 16#ffffff}   :: {0..255, 0..16#ffffff},
```
[the remark this answers is not in the document]
Comment (RoadRunnr) [reply include/ergw.hrl]: yes, only 1, 2 and 3 are actually defined. the rest is valid but not used.
1 stands for "normal traffic"

File: src/ergw_inet.erl [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 -module(ergw_inet).
 
--export([ip2bin/1, bin2ip/1, ipv6_interface_id/2, ipv6_interface_id/3]).
+-export([ip2bin/1, bin2ip/1, ipv6_interface_id/2, ipv6_interface_id/3,
+	 ipv6_prefix/1, ipv6_prefix/2]).
```
[the remark this answers is not in the document]
Comment (RoadRunnr) [reply src/ergw_inet.erl]: I don't understand the question.

What other functions should/could be added to that export?

File: src/ergw_http_api.erl [hunk 3]
```diff
@@ -59,22 +59,29 @@ start_http_listener(#{ip := IP, port := Port, acceptors_num := AcceptorsNum}) ->
 		   {acceptors_num, 100}]).
```
[the remark this answers is not in the document]
Comment (RoadRunnr) [reply src/ergw_http_api.erl]: yes, i'll change that

File: src/ergw_inet.erl (hunk 2 above)
[the remark this answers is not in the document]
Comment (ebengt) [reply src/ergw_inet.erl]: There are two exports. The new functions are added to one of the exports. I wondered why this export (and not the other).
There is a difference between the exports and I am interested in finding out that difference.

[the remark this answers is not in the document]
Comment (RoadRunnr) [reply src/ergw_inet.erl]: ah, the second export is for functions that deal with stuff related to generating raw IP packets. The other functions deal with formatting and converting IP addresses.

**Pull request merged**